packages/kotti-ui/source/kotti-field/hooks.ts

```diff
@@ -348,3 +350,72 @@ export const useForceUpdate = (): {
 		forceUpdateKey,
 	}
 }
+
+export const useKtFieldRef = (): Ref<{
+	inputContainerRef: HTMLDivElement
+	inputContainerWrapperRef: HTMLDivElement
+} | null> => {
+	/**
+	 * both inputContainerRef and inputContainerWrapperRef
+	 * are template refs on KtField.vue
+	 */
+	const ktFieldRef = ref<{
+		inputContainerRef: Ref<HTMLDivElement>
+		inputContainerWrapperRef: Ref<HTMLDivElement>
+	} | null>(null)
+
+	return ktFieldRef
+}
+
+export const useEmitBlur = <T>({
+	emit,
+	field,
+	fieldTarget,
+	findEventTarget,
+	valuePlaceholder,
+}: {
+	emit: (event: 'blur', value: T | null) => void
+	field: KottiField.Hook.Returns<T>
+	fieldTarget: Ref<Array<HTMLElement | null> | HTMLElement | null>
+	findEventTarget?: (target: EventTarget | null) => EventTarget | null
+	valuePlaceholder?: T | null
```

Comment:
`valuePlaceholder` is not the best name. a placeholder would be something that would be given, if the original value is null, but this does the opposite. I'd call this `valueOverride` or `valueReplacement`